### PR TITLE
improve memory leak test

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

**tldr: Uncovered a memory leak test bug that causes it to pass/fail unreliably. I may get this merged asap so that our CI goes back to reliable + green for any weekend PRs.**

A bit of fun to end the day: I was seeing some flakiness in the memory test and ended up refactoring a bit of it. There's a mix of usability changes (`--ci` mode if you want a final pass/fail, otherwise it will run forever and let you spot the trend in the output over time) and actual test reliability improvements. Mainly:

- Added `--expose-gc` support so that we can guarantee that garbage collection runs
- Switched `total_head_size -> used_heap_size`. `total` is the total reserved, which is largely up to Node.js. Used is actual memory used by the application. 

With these two changes I now see reliable results across every run, locally.

```
$ node scripts/memory/index.js
ERROR: Node must be run with --expose-gc
```         

```
$ node --expose-gc scripts/memory/index.js
Greetings, human. This test will run forever. Run with the "--ci" flag to finish with a result.
Snapshot 001: 106.69290329060044%
Snapshot 002: 106.3646071611346%
Snapshot 003: 106.62807742564124%
Snapshot 004: 106.76272397910616%
Snapshot 005: 107.130898566085%
...
Snapshot 121: 109.18502642715731%
Snapshot 122: 109.2200169022692%
Snapshot 123: 109.21072172262116%
```

```
$ node --expose-gc scripts/memory/index.js --ci
Estimate 000/100: 155.98816145873386%
Estimate 001/100: 144.71746216879%
Estimate 002/100: 129.19713350080747%
...
Estimate 097/100: 150.28960848649527%
Estimate 098/100: 140.2301594422086%
Estimate 099/100: 169.27028594498987%
Test complete. Running final garbage collection...

Result: PASS (110.47501025351825%)
Memory usage began at 33.1 MB and finished at 33.1 MB.
The threshold for a probable memory leak is 120%

Exiting...
```

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
Written by someone who clearly has scars from past memory bugs! Looks great.
